components/net/websocket_loader.rs

```diff
@@ -70,96 +114,52 @@ pub fn init(connect: WebSocketCommunicate, connect_data: WebSocketConnectData, c
 
         // URL that we actually fetch from the network, after applying the replacements
         // specified in the hosts file.
-        let net_url_result = parse_url(replace_hosts(&connect_data.resource_url).as_url().unwrap());
-        let net_url = match net_url_result {
-            Ok(net_url) => net_url,
-            Err(e) => {
-                debug!("Failed to establish a WebSocket connection: {:?}", e);
-                let _ = connect.event_sender.send(WebSocketNetworkEvent::Fail);
-                return;
-            }
-        };
-        let channel = establish_a_websocket_connection(&connect_data.resource_url,
-                                                       net_url,
-                                                       connect_data.origin,
-                                                       connect_data.protocols.clone(),
-                                                       cookie_jar);
-        let (_, ws_sender, mut receiver) = match channel {
-            Ok(channel) => {
-                let _ = connect.event_sender.send(WebSocketNetworkEvent::ConnectionEstablished(channel.0.clone(),
-                                                                                               connect_data.protocols));
-                channel
-            },
-            Err(e) => {
-                debug!("Failed to establish a WebSocket connection: {:?}", e);
-                let _ = connect.event_sender.send(WebSocketNetworkEvent::Fail);
-                return;
-            }
+        let net_url = replace_hosts(&connect_data.resource_url).into_url().unwrap();
 
+        let host = Host {
+            hostname: connect_data.resource_url.host_str().unwrap().to_owned(),
+            port: connect_data.resource_url.port_or_known_default(),
         };
 
-        let initiated_close = Arc::new(AtomicBool::new(false));
```

Comment:
Why don't we need this any more? It was added in 27b4ac61508877e6ded6f01a65bd02e01dc4e81b to address the case where we should not echo the server's close message if we initiated the closure.